FLT/DivisionAlgebra/ModuleTopProd.lean

```diff
@@ -0,0 +1,355 @@
+import Mathlib.Topology.Algebra.Module.ModuleTopology
+
+namespace ModuleTopIsProd
+
+variable (R S : Type*) [Ring R] [Ring S] [TopologicalSpace R] [TopologicalSpace S] (M N : Type*)
+  [AddCommMonoid M] [AddCommMonoid N] [Module R M] [Module S N] [TopologicalSpace M]
+  [IsModuleTopology R M] [TopologicalSpace N] [IsModuleTopology S N]
+
+instance : HSMul (R × S) (M × N) (M × N) where
+  hSMul a b := (a.1 • b.1, a.2 • b.2) -- why is this needed...
```

Comment:
Just define SMul rather than HSMul. And I think you should not make this a global instance because it causes diamonds [#mathlib4 > product of modules over product of rings @ 💬](https://leanprover.zulipchat.com/#narrow/channel/287929-mathlib4/topic/product.20of.20modules.20over.20product.20of.20rings/near/532797383) (this is why it's not already there; you might want to look through the Zulip thread. 

What I suggest you do is to instead make this a definition, give it a name (you can use `#whatsnew in` to find out the name which Lean automatically gave the instance and if it's not too ridiculous you can use it) and then make it a local instance in this file only (and in other files where you need to have it).